polar-core/src/vm.rs

```diff
@@ -1418,16 +1412,18 @@ impl PolarVirtualMachine {
                 return self.query_for_operation(term);
             }
             Value::Variable(sym) => {
-                if let VariableState::Bound(val) = self.variable_state(sym) {
-                    self.push_goal(Goal::Query { term: val })?;
-                } else {
-                    // variable was unbound
-                    // apply a constraint to variable that it must be truthy
-                    self.push_goal(Goal::Unify {
-                        left: term.clone(),
-                        right: term!(true),
-                    })?;
-                }
+                self.push_goal(
```

Comment:
I'm not really sure this is easier to read if that was the rationale behind this change.